grakn-graph/src/main/java/ai/grakn/graph/internal/TypeImpl.java

```diff
@@ -119,7 +119,7 @@ public void txCacheClear(){
      * @return A new or already existing instance
      */
     V putInstance(Schema.BaseType instanceBaseType, Supplier<V> finder, BiFunction<VertexElement, T, V> producer) {
-        vertex().graph().checkMutationAllowed();
+        preCheckForInstanceCreation();
```

Comment:
Yes and I always was because `addInstance` gets called exclusively as well. I might be able to fix that though. Hold on. . . .